server/graphql/v2/mutation/AddedFundsMutations.ts

```diff
@@ -0,0 +1,352 @@
+      },
+      memo: {
+        type: GraphQLString,
+        description: 'A private note for the host',
+      },
+      processedAt: {
+        type: GraphQLDateTime,
+        description: 'The date at which the order was processed',
+      },
+      invoiceTemplate: {
+        type: GraphQLString,
+        description: 'The invoice template to use for this order',
+      },
+      tax: {
+        type: GraphQLTaxInput,
+        description: 'The tax to apply to the order',
+      },
+      accountingCategory: {
+        type: GraphQLAccountingCategoryReferenceInput,
+        description: 'The accounting category of this order',
+      },
+    },
+    resolve: async (
+      _,
+      args: Omit<AddFundsMutationArgs, 'transactionsImportRow'> & { order: ObjectReference },
+      req: express.Request,
+    ) => {
+      checkRemoteUserCanUseHost(req);
+
+      const order = await fetchOrderWithReference(args.order, { throwIfMissing: true });
+      const account: Collective = await fetchAccountWithReference(args.account, { throwIfMissing: true });
+      const fromAccount: Collective = await fetchAccountWithReference(args.fromAccount, { throwIfMissing: true });
+      const tier = args.tier && (await fetchTierWithReference(args.tier, { throwIfMissing: true }));
+      const host = await account.getHostCollective({ loaders: req.loaders });
+
+      // Check amounts
+      const totalAmount = getValueInCentsFromAmountInput(args.amount, { expectedCurrency: account.currency });
+      const paymentProcessorFee = args.paymentProcessorFee
+        ? getValueInCentsFromAmountInput(args.paymentProcessorFee, { expectedCurrency: account.currency })
+        : 0;
+
+      validateAddFundsArgs({ account, fromAccount, args, paymentProcessorFee, totalAmount, host, req });
+
+      await twoFactorAuthLib.enforceForAccount(req, host, { onlyAskOnLogin: true });
+
+      const accountingCategory =
+        args.accountingCategory &&
+        (await fetchAccountingCategoryWithReference(args.accountingCategory, {
+          throwIfMissing: true,
+          loaders: req.loaders,
+        }));
+
+      // Refund Existing Order
+      const transactions = await order.getTransactions({ order: [['id', 'desc']] });
+      assert(transactions.length > 0, 'No ADDED FUNDS transaction found for this order');
+      const editedTransactions = transactions.map(t => t.id);
+      await Promise.all(transactions.map(transaction => transaction.destroy()));
+
+      const previousData = order.toJSON();
+      // Update existing Order
+      await order.update({
+        CreatedByUserId: req.remoteUser.id,
+        FromCollectiveId: fromAccount.id,
+        CollectiveId: account.id,
+        totalAmount: totalAmount,
+        currency: account.currency,
+        description: args.description || order.description,
+        status: OrderStatuses.NEW,
+        TierId: tier?.id || order.TierId,
+        AccountingCategoryId: accountingCategory?.id || order.AccountingCategoryId,
```

Comment:
:no_entry_sign: We must make sure that the accounting cateogory belongs to the host first. For add funds, this is done in the `common` lib function, by relying on `checkCanUseAccountingCategoryForOrder`.